packages/emotion-primitives/src/createEmotion.js

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import reactPrimitives from 'react-primitives'
+import PropTypes from 'prop-types'
+
+import { getStyles } from './getStyles'
+import { convertToRNStyles } from './convertToRNStyles'
+
+const isValidPrimitive = primitive =>
+  ['Text', 'View', 'Image'].indexOf(primitive) > -1
+
+const getPrimitive = primitive => {
+  if (typeof primitive === 'string' && isValidPrimitive(primitive)) {
+    return reactPrimitives[primitive]
+  } else if (typeof primitive === 'string' && !isValidPrimitive(primitive)) {
+    throw new Error(
+      `Cannot style invalid primitive ${primitive}. Expected primitive to be one of ['Text', 'View', 'Image']`
+    )
+  } else if (typeof primitive === 'function') {
+    return primitive
+  }
+}
+
+function evalStyles(context, Comp, styles, styleOverrides) {
+  // Assign static property so that the styles can be reused (like in withComponent)
+  Comp.styles = convertToRNStyles.call(context, styles)
+
+  return getStyles.call(context, Comp.styles, context.props, styleOverrides)
+}
+
+/**
+ * Creates a function that renders the styles on multiple targets with same code.
+ */
+export function createEmotionPrimitive(splitProps) {
+  /*
+   * Returns styled component
+   */
+  return function emotion(primitive, { displayName } = {}) {
```

Comment:
passing `displayName` like that is not part of react-emotion's API, do we have to do it here?